etna/collections/models.py

```diff
@@ -205,6 +217,16 @@ def results_pages(self):
     ]
     subpage_types = ["collections.TimePeriodExplorerPage", "collections.ResultsPage"]
 
+    def get_context(self, request):
+        context = super().get_context(request)
+        context["insights_time_periods"] = (
+            InsightsPage.objects.filter(time_period=self)
+            .live()
+            .order_by("title")
+            .specific()[:3]
```

Comment:
You only need to use `specific()` when starting the queryset from the generic `Page` model manager, e.g. `Page.objects.all()`. This queryset will already be 'specific', because you're using the `InsightsPage` model manager.